Set use_v2_api.series to true by default

djmitche opened the pull request:
This will enable the v2 API for all customers once they upgrade to 7.39.

### Motivation

V2 API is working well internally.

### Describe how to test/QA your changes

No need.  This is already enabled internally.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

djmitche committed:
Set use_v2_api.series to true by default

This will enable the v2 API for all customers once they upgrade to 7.39.

olivielpeau reviewed (commented):
I haven't followed the rollout closely enough to be able to approve/reject this, but I have a few questions/comments:

* the PR title says the opposite of what the PR does :)
* do we have any test in place that checks that setting `enable_payloads.series` to `false` still disables series payloads when the v2 series API is used?
* could you confirm that this change is approved from metrics-intake's perspective? Please ask them to approve this change. It's important that they're aware of the expected rate at which we expect Datadog customers to upgrade to the newer version of the Agent that will include this, so they can do some capacity planning (if any is needed).

djmitche commented:
> the PR title says the opposite of what the PR does :)

Oops! the commit title, too.  fixed.

> do we have any test in place that checks that setting enable_payloads.series to false still disables series payloads when the v2 series API is used?

Yes: `TestSendV1Series`

> could you confirm that this change is approved from metrics-intake's perspective

@lliissoonngg asked me to make this change.  I've added him as an approver.

lliissoonngg reviewed (approved):
This flag has been turned on for a while: https://github.com/DataDog/k8s-datadog-agent-ops/blob/main/k8s/datadog-agent/values/prod.yaml#L39-L40

djmitche committed:
Update releasenotes/notes/use_v2_api_for_series-53ac2d6b49b181b9.yaml

Co-authored-by: Bryce Eadie <bryce.eadie@datadoghq.com>

lliissoonngg commented:
Add a small doc so that we won't forget:-)

https://docs.google.com/document/d/1EtNxCoyak1bMmEBELVvZG4MY01vp3oIvTh5Ff4faNXA/edit

djmitche commented:
This is going to wait for 7.40, and is a trivial PR, so I'll close it for now.

carlosroman commented:
This PR is now ready for `7.41`.

djmitche reviewed (commented):
I can't approve since I originally filed the PR, but here's a 👍🏽 from ASC
